### PR TITLE
[consensus] init commit for fuzzing proposals messages

### PR DESCRIPTION
**Pull request by mimoo**
**MOTIVATION:**

This PR sets up two things:

* an `event_processor_fuzzing.rs` next to `event_processor_test.rs`. The file contains a necessary function for a fuzzer, and a test so that breaking changes in EventProcessor won't break the fuzzer.
* calls from our test suite's Libra fuzzer.

The fuzzer is not finished, but this brings the foundation we can work on to finish the fuzzer.

**TEST PLAN:**

```
cargo test -p consensus
```

### CODE REVIEW

**Comment by mimoo** [comment]
addressed all comments!

**Comment by mimoo** [comment]
rebased

**Comment by mimoo** [comment]
blocked by `#[global_allocator]` defined by jemalloc.

File: consensus/src/chained_bft/event_processor.rs [hunk 1]
```diff
@@ -134,31 +138,42 @@ impl<T: Payload> EventProcessor<T> {
...
         };
+        // are we the current round's leader?
         if self
             .proposer_election
             .is_valid_proposer(self.author, new_round_event.round)
             .is_none()
         {
             return;
         }
+        // generate the proposal
```
Comment (zekun000) [inline consensus/src/chained_bft/event_processor.rs]: those comments look redundant, can we remove them? the code is pretty self-explanatory

Comment (zekun000) [reply consensus/src/chained_bft/event_processor.rs]: I didn't comment on all of them, but I'm referring to all those just repeat the next line's function name

Comment (mimoo) [reply consensus/src/chained_bft/event_processor.rs]: I don't really agree with this. I think it's good to be able to understand what a function does from a cursory glance, especially critical code. But in the interest of time I removed them : o

**Pull request merged**